File: modules/core/src/effects/lighting/ambient-light.js
const DEFAULT_LIGHT_COLOR = [255, 255, 255];
const DEFAULT_LIGHT_INTENSITY = 1.0;

let idCount = 0;

export class AmbientLight {
  constructor(props = {}) {
    const {color = DEFAULT_LIGHT_COLOR} = props;
    const {intensity = DEFAULT_LIGHT_INTENSITY} = props;

    this.id = props.id || `ambient-${idCount++}`;
    this.color = color;
    this.intensity = intensity;
    this.type = 'ambient';
  }
}

File: bindings/pydeck/examples/lighting/ambient_light.py
"""AmbientLight
===============

Illuminate every surface evenly with ambient light.
"""

import pydeck as pdk

from _utils import write_lighting_example

write_lighting_example(
    pdk.Effect("AmbientLight", color=[170, 210, 255], intensity=1.2),
    "ambient_light.html",
)
